pyccel/stdlib/math/pyc_math_c.h

```diff
@@ -13,8 +13,11 @@
 ** operands of modulo operator.
 */
 
-#define MOD_PYC(N, M) ((N < 0 ^ M < 0) && (N % M ) != 0 ? (N % M) + M : (N % M))
-#define FMOD_PYC(N, M) ((N < 0 ^ M < 0) && fmod(N, M) != 0  ? fmod(N, M) + M : fmod(N, M))
+#define MOD_PYC(N, M) \
+    ({long int RES_MOD = (N % M); ((N < 0 ^ M < 0) && (RES_MOD ) != 0) ? (RES_MOD) + M : (RES_MOD);})
```

Comment:
Thanks for the changes.

I think there is still a problem with the parentheses. One is missing, and several are useless:

```suggestion
#define MOD_PYC(N, M) \
    ({long int RES_MOD = N % M; ((N < 0 ^ M < 0) && RES_MOD != 0 ? RES_MOD + M : RES_MOD);})
```

Reply:
I also find that it is more natural to test for equality, rather than testing for inequality. What do you think?

Reply:
I am surprised that all tests pass, given the parenthesis problem... Are you 100% sure that you are properly testing this function?

Reply:
All my tests are added to the file "test_epyccel_mod.py" and i think i added all the cases to test the functionality of this function

Reply:
I will check again this function, fix the parenthesis problem and also see if I can test for the equality first .

Reply:
Does this need to be a macro? If it is sufficiently complicated to need additional variables would an inline function be more pertinent?

Reply:
I thought of it, but I wasn't sure if this was the right thing to do since we are going to add just one variable, using the inline function will make the code more readable, so I am thinking of changing the macro into an inline function.

Reply:
> @EmilyBourne If you write it as a function, how do you make sure that the compiler inlines it?

we can use the ```inline``` function specifier for C

Reply:
So you want to declare it as `inline` within the header itself?

Reply:
If we declare it as `extern`, I don't think that it will be inlined by the linker, unless we pass other flags to it. But I am fine with having the inline function declaration in the header.

Reply:
@soukainahikma Given the discussion above, please change these macros into `inline` functions. At the same time you could do the same with the functions `pyc_radians` and `pyc_degrees` which are currently in the file `pyccel/stdlib/math/pyc_math_c.c`.

Reply:
> > we can use the inline function specifier for C
>
> So you want to declare it as inline within the header itself?

That is what I was suggesting yes. This should make it easier to read the parentheses too

Reply:
we can't have only declaration in the header because the specifier inline will make it only available in the header file

Reply:
if we add this to the header file
```C
inline int ab(int a, int b)
{
        return a + b;
}
```
and called it from another even if the header is included there we will get `undefined reference`

Reply:
after discussing with @EmilyBourne we can use the `inline` specifier together with the `extern` specifier to define a function in a header file.

@soukainahikma for this issue you can add the inlined functions in the header file instead of the Macros using the example below.

ex:
```C
//function definition in Header file
extern inline func(void)
{
//body
}
```


<details>
 <summary>C standard about inline specifier</summary>

```
A function specifier may appear more than once; the behavior is the same as if it
appeared only once.
A function declared with an inline function specifier is an inline function. Making a
function an inline function suggests that calls to the function be as fast as possible.
Any function with internal linkage can be an inline function. For a function with external
linkage, the following restrictions apply: If a function is declared with an inline
function specifier, then it shall also be defined in the same translation unit. If all of the
file scope declarations for a function in a translation unit include the inline function
specifier without extern, then the definition in that translation unit is an inline
definition. An inline definition does not provide an external definition for the function,
and does not forbid an external definition in another translation unit. An inline definition
provides an alternative to an external definition, which a translator may use to implement
any call to the function in the same translation unit. It is unspecified whether a call to the
function uses the inline definition or the external definition.
```

</details>

@yguclu the standard state that the inline specifier will not always inline the function but will always try to call the function the fastest way possible.